src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -563,18 +564,46 @@ public static void _stopJenkins(Server server, List<LenientRunnable> tearDowns,
                 }
             }
         }
-
-        if (jenkins != null) {
-            jenkins.cleanUp();
+        try (var ignored = new TemporaryConsoleLogTweak("hudson.XmlFile", Level.FINEST)) {
+            if (jenkins != null) {
+                jenkins.cleanUp();
+            }
+            ExtensionList.clearLegacyInstances();
+            DescriptorExtensionList.clearLegacyInstances();
         }
-        ExtensionList.clearLegacyInstances();
-        DescriptorExtensionList.clearLegacyInstances();
-
         if (exception.getSuppressed().length > 0) {
             throw exception;
         }
     }
 
+    private static final class TemporaryConsoleLogTweak implements AutoCloseable {
```

Comment:
Why is this temporary, and how will we ensure that it does not become permanent?

Reply:
This is not meant to be temporary, I renamed the class to clarify the intent.

Reply:
> This is not meant to be temporary

What would its value be after we diagnose and fix the bug seen in https://github.com/jenkinsci/jenkins/pull/8511/checks?check_run_id=17000216956?

Reply:
If there is no more race condition, I would fine removing it, but the thing is with this kind of issue is that it always pop up at an unexpected time and you often don't have enough data to investigate. On the other hand its impact is very light (under normal condition this adds one stacktrace writing `queue.xml`) since you wouldn't expect much xmlfile write being donc between stopping the test Jenkins instance and the cleanup sequence.

Reply:
Fine with the merge of this PR if and only if it is reverted once its diagnostic value has been exhausted.